tests/integration_tests/data_steward/gcloud/gcs_client_test.py

```diff
@@ -25,36 +25,77 @@ def setUp(self):
                                     f'{self.prefix}/c', f'{self.prefix}/d')
+        self.client.empty_bucket(self.bucket_name)
+        self._stage_bucket()
+
+    def test_get_bucket_items_metadata(self):
+
+        items_metadata: list = self.client.get_bucket_items_metadata(
+            self.bucket_name)
+
+        actual_metadata: list = [item['name'] for item in items_metadata]
+        expected_metadata: list = [
+            f'{prefix}/obj.txt' for prefix in self.sub_prefixes
+        ]
+
+        self.assertCountEqual(actual_metadata, expected_metadata)
+        self.assertIsNotNone(items_metadata[0]['id'])
+
+    def test_get_blob_metadata(self):
+
+        bucket = self.client.get_bucket(self.bucket_name)
+        blob_name: str = f'{self.sub_prefixes[0]}/obj.txt'
+
+        blob = bucket.blob(blob_name)
+        metadata: dict = self.client.get_blob_metadata(blob)
+
+        self.assertIsNotNone(metadata['id'])
+        self.assertIsNotNone(metadata['name'])
+        self.assertIsNotNone(metadata['bucket'])
+        self.assertIsNotNone(metadata['generation'])
+        self.assertIsNotNone(metadata['metageneration'])
+        self.assertIsNotNone(metadata['contentType'])
+        self.assertIsNotNone(metadata['storageClass'])
+        self.assertIsNotNone(metadata['size'])
+        self.assertIsNotNone(metadata['md5Hash'])
+        self.assertIsNotNone(metadata['crc32c'])
+        self.assertIsNotNone(metadata['etag'])
+        self.assertIsNotNone(metadata['updated'])
+        self.assertIsNotNone(metadata['timeCreated'])
+
+        self.assertEqual(metadata['name'], blob_name)
+        self.assertEqual(metadata['size'], len(self.data))
 
     def test_empty_bucket(self):
 
         self.client.empty_bucket(self.bucket_name)
-        self._stage_bucket()
-        self.client.empty_bucket(self.bucket_name)
-
-        actual = self.client.list_blobs(self.bucket_name)
-        expected: list = []
+        items: list = self.client.list_blobs(self.bucket_name)
 
-        self.assertCountEqual(actual, expected)
+        # check that bucket is empty
+        self.assertCountEqual(items, [])
 
     def test_list_sub_prefixes(self):
-        self.client.empty_bucket(self.bucket_name)
-        self._stage_bucket()
 
-        items = self.client.list_sub_prefixes(self.bucket_name, self.prefix)
+        items: list = self.client.list_sub_prefixes(self.bucket_name,
+                                                    self.prefix)
+        sorted_items: list = sorted(items)
 
-        self.assertEqual(len(self.sub_prefixes), len(items))
-        for item in items:
-            self.assertIn(item[:-1], self.sub_prefixes)
+        # Check same number of elements
+        self.assertEqual(len(self.sub_prefixes), len(sorted_items))
 
-        self.client.empty_bucket(self.bucket_name)
+        # Check same prefix
+        for index, item in enumerate(sorted_items):
+            self.assertEqual(item[:-1], self.sub_prefixes[index])
```

Comment:
Do you think this sorting can be done away with using the same `assertCountEqual` check in the other test?

Reply:
yes.  It can.  Removing and pushing up.  Good catch as usual.